### ISSUE TITLE
unable to load lib

### ISSUE THREAD

**Issue by valeriob01** [preda/gpuowl#112]
./gpuowl -h
./gpuowl: error while loading shared libraries: libOpenCL.so.1: cannot open shared object file: No such file or directory

but the file is present.
ROCm 3.1.0

**Comment by preda** [comment]
export LD_LIBRARY_PATH ?

**Comment by valeriob01** [comment]
> export LD_LIBRARY_PATH ?

already doing that but doesn't solve the problem.

**Comment by preda** [comment]
"ldd gpuowl" to see what dynamic libs it's looking for, and where they are found.
Anyway I don't understand why gpuowl is looking for "libOpenCL.so.1", as it's supposed to be linking against libOpenCL.so (without version).

**Comment by valeriob01** [comment]
> "ldd gpuowl" to see what dynamic libs it's looking for, and where they are found.
> Anyway I don't understand why gpuowl is looking for "libOpenCL.so.1", as it's supposed to be linking against libOpenCL.so (without version).

There is a symbolic link that probably gpuowl does not like.

**Comment by valeriob01** [comment]
> > "ldd gpuowl" to see what dynamic libs it's looking for, and where they are found.
> > Anyway I don't understand why gpuowl is looking for "libOpenCL.so.1", as it's supposed to be linking against libOpenCL.so (without version).
> 
> There is a symbolic link that probably gpuowl does not like.

it is a soft link:

libOpenCL.so -> libOpenCL.so.1

**Comment by valeriob01** [comment]
> "ldd gpuowl" to see what dynamic libs it's looking for, and where they are found.
> Anyway I don't understand why gpuowl is looking for "libOpenCL.so.1", as it's supposed to be linking against libOpenCL.so (without version).

ldd gpuowl
	linux-vdso.so.1 (0x00007ffcff1ae000)
	libOpenCL.so.1 => /opt/rocm/opencl/lib/x86_64/libOpenCL.so.1 (0x00007fdd4cc61000)
	libgmp.so.10 => /lib/x86_64-linux-gnu/libgmp.so.10 (0x00007fdd4cbde000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fdd4ca5a000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fdd4c8d7000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fdd4c8bd000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fdd4c89c000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fdd4c6d9000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fdd4c6cf000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fdd4c6ca000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fdd4cd12000)

**Comment by valeriob01** [comment]
> > > "ldd gpuowl" to see what dynamic libs it's looking for, and where they are found.
> > > Anyway I don't understand why gpuowl is looking for "libOpenCL.so.1", as it's supposed to be linking against libOpenCL.so (without version).
> > 
> > 
> > There is a symbolic link that probably gpuowl does not like.
> 
> it is a soft link:
> 
> libOpenCL.so -> libOpenCL.so.1

```
/opt/rocm/opencl/lib/x86_64# ls -la
total 84856
drwxrwxr-x 3 1001 1001     4096 Mar 14 08:21 .
drwxrwxr-x 3 1001 1001     4096 Mar 14 08:21 ..
drwxrwxr-x 2 1001 1001     4096 Mar 14 08:21 bitcode
-rwxrwxr-x 1 1001 1001 82582128 Nov 19 22:58 libamd_comgr.so
-rwxrwxr-x 1 1001 1001  3515264 Nov 19 22:58 libamdocl64.so
-rwxrwxr-x 1 1001 1001   737944 Nov 19 22:58 libcltrace.so
lrwxrwxrwx 1 1001 1001       14 Nov 19 23:00 libOpenCL.so -> libOpenCL.so.1
-rwxrwxr-x 1 1001 1001    34920 Nov 19 22:58 libOpenCL.so.1

```

**Comment by preda** [comment]
Do you still have this problem, or is it fixed? Does it happen with ROCm 2.10 or 3.1? Since when does it happen?

**Comment by valeriob01** [comment]
> Do you still have this problem, or is it fixed? Does it happen with ROCm 2.10 or 3.1? Since when does it happen?

Thank you for replying. This happened with ROCm 3.1 and not with ver. 2.10 which I still use.

**Comment by preda** [comment]
Closing for now.
